test/junit/scala/collection/immutable/HashMap1Test.scala

```diff
@@ -0,0 +1,29 @@
+package scala.collection.immutable
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import java.io.{ByteArrayOutputStream, ObjectOutputStream, ByteArrayInputStream, ObjectInputStream}
+
+@RunWith(classOf[JUnit4])
+class HashMap1Test {
+
+  @Test
+  def deserializesCorrectly() {
+  	val kv = (1->1)
+    val map = new HashMap.HashMap1(kv._1,kv._1.hashCode,kv._2,kv)
+
+ 	val baos = new ByteArrayOutputStream()
+    val oos = new ObjectOutputStream(baos)
+    oos.writeObject(map)
+    oos.close()
+    
+    val bais = new ByteArrayInputStream(baos.toByteArray())
+    val ois = new ObjectInputStream(bais)
+    val deserializedMap = ois.readObject().asInstanceOf[HashMap$HashMap1]
+    ois.close()
+    
+    assertEquals(kv, deserializedMap.kv)
```

Comment:
This could be an opportunity to refactor serialisation test code into a utilities object. There's a few places where this is done. This package looks like a suitable location,

![scala-testing](https://cloud.githubusercontent.com/assets/1123855/13550878/9d2f2740-e31f-11e5-95cc-ec039094c414.png)

Here's one example of where that could be used,

![scala-testing-example](https://cloud.githubusercontent.com/assets/1123855/13550886/c5c0eb8a-e31f-11e5-9d35-5b5f4344e81f.png)

I am not suggesting you expand the scope of this commit beyond moving the ser/des code into a shared location but it could be good to at least put the code there as part of this fix and then return in a later commit to replace other occurrences of the ser/des pattern with the common code. WDYT?